packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -512,7 +516,38 @@ public void onCreate(@NonNull LifecycleOwner owner) {
     if (disposed) {
       return;
     }
-    mapView.onCreate(null);
+    if (owner instanceof SavedStateRegistryOwner) {
+      // Accessing SavedStateRegistry *must* be done in DefaultLifecycleObserver#onCreate for
+      // reasons described in the javadoc of GoogleMapsPlugin.StateSavingLifecycleProvider.
+      SavedStateRegistry savedStateRegistry =
+          ((SavedStateRegistryOwner) owner).getSavedStateRegistry();
+      String key = SAVED_STATE_KEY + id;
```

Comment:
Mmmm these IDs are assigned in order of platform view creation.
To my understanding: if the app is coming back to a restored state the key may match a different view. e.g:

Let's say we have an app with 2 pages (pageA, pageB), and a map on each page (mapA, mapB). The app starts at pageA, and can then transition to pageB.

The ID for mapA would be 1 and form mapB will be 2.
Let's say the activity is cleared from memory when in mapB, when restoring the instance state - if the app is restoring back to pageB, now mapB will have id 1 and the wrong state will be restored.

I'm not exactly sure what will be the best approach here, I guess we'd need to use some sort of persistent widget identifier defined by the app developer?

cc @goderbauer (master of state restorations), is what I'm describing correct? assuming all the widgets in the app are restorable if we're coming back directly to pageB the widgets of pageA won't be created first correct? any advice here?

Reply:
Ah that's a great point, I didn't realize that's how the IDs work. That would certainly create a wrinkle in this design. I'll wait for a reply from @goderbauer for further guidance.

Reply:
There's no garantee that the widgets will be created in the same order again. In the state restoration framework for the flutter framework we require developers to assign stable `restorationId`s to widgets or properties that they need to restore. The ID needs to be stable across app launches.

Reply:
Is there any way for the PlatformView to get access to that ID? If multiple maps are used, state saving needs a unique stable ID for each map.

Reply:
The widget in question would have to take a restoration ID as an argument and send it over to this size here.

Reply:
Got it. Since a situation with 2 maps should be rare, I can definitely add that as an optional argument and avoid breaking API changes.